src/js/base/amd_loader.js

```diff
@@ -0,0 +1,149 @@
+var define, requirejs;
+(function () {
+    var isArray = function(arg) {
+        return Object.prototype.toString.call(arg) === '[object Array]';
+    };
+
+    var nodeRequire = (typeof require != "undefined") ? require : null;
+
+    // module name -> module object
+    var moduleTable = {};
+
+    function moduleResolved(modName) {
+        return (modName in moduleTable) && (moduleTable[modName].resolved);
+    }
+
+    function buildCallbackArgs(deps) {
+        return deps.map(function(depName) {
+            if (moduleResolved(depName)) {
+                return moduleTable[depName].val;
+            } else {
+                throw new Error("Module " + depName + " not yet resolved");
+            }
+        });
+    }
+
+    function evaluateModuleFn(moduleObj) {
+        console.log("Evaluating " + moduleObj.name);
+
+        if (moduleObj.val != null) {
+            throw new Error("Already evaluated " + moduleObj.name);
+        }
+
+        var callbackArgs = buildCallbackArgs(moduleObj.deps);
+
+        // Now call it
+        moduleObj.val = moduleObj.callback.apply(null, callbackArgs);
```

Comment:
NOTE (add comment): This should call out explicitly that no stack management is needed, because this happens long before any Pyret code runs – this is for pure JS dependencies.